### ISSUE TITLE
feature:  postgres integration

### ISSUE THREAD

**Issue by Davo1624** [slskd/slskd#690]
Rather than use the default database, allow connections to postgres db

**Comment by jpdillingham** [comment]
Thanks for raising this.  I'm trying to keep this in mind as I work on new features that use database connections so hopefully it is fairly easy to do when the time comes.

The databases other than the ones used for shares should be pretty straightforward to implement, as they are using Entity Framework as the data access layer.  The shares stuff is very SQLite specific, but might not have to be.

**Comment by SquaredPotato** [comment]
Hello!
First of all, thanks for maintaining this project!
Are there any plans on doing this? I notice that the transfers database tends to temporarily lock up when someone queues a lot of files (>1000), causing other transfers to error due to timeouts.
This is probably due to me using longhorn for general storage, but as I am already hosting cloudnative postgres as well, using a much faster storage backend, I was wondering if this had any priority.

**Comment by jpdillingham** [comment]
No immediate plans to implement this, but I am probably going to end up splitting active/queued transfers from completed ones, and will keep the former in an in-memory SQLite database, then move the records somewhere else.  Once this is done it would be fairly straightforward to also replicate those records to an external database, but I'm not sure it would be as beneficial.

And yes your problem is probably the filesystem; anything that uses 'copy-on-write' or is in any way 'eventually consistent' will cause problems.

**Comment by SquaredPotato** [comment]
Yep, I switched to openebs/local-hostpath for slskd's config folder and that fixes the issue as well :)
